algorithms/kernel/k_nearest_neighbors/bf_knn_impl.i

```diff
@@ -107,34 +106,55 @@ public:
-                                          DAAL_UINT64 resultsToCompute, const size_t nClasses, const size_t k, VoteWeights voteWeights,
-                                          int * trainLabel, const NumericTable * trainTable, const NumericTable * testTable,
-                                          NumericTable * testLabelTable, NumericTable * indicesTable, NumericTable * distancesTable,
-                                          TlsMem<FPType, cpu> & tlsDistances, TlsMem<int, cpu> & tlsIdx, TlsMem<FPType, cpu> & tlsMaxs,
-                                          TlsMem<FPType, cpu> & tlsKDistances, TlsMem<int, cpu> & tlsKIndexes, TlsMem<int, cpu> & tlsVoting)
+    struct BruteForceTask
     {
-        FPType * distancesBuff = tlsDistances.local();
-        DAAL_CHECK_MALLOC(distancesBuff);
+    public:
+        DAAL_NEW_DELETE();
+        FPType * kDistances;
+        FPType * maxs;
+        int * kIndexes;
 
-        int * idx = tlsIdx.local();
-        DAAL_CHECK_MALLOC(idx);
+        static BruteForceTask * create(const size_t inBlockSize, const size_t outBlockSize, const size_t k)
+        {
+            auto object = new BruteForceTask(inBlockSize, outBlockSize, k);
+            if (object && object->isValid()) return object;
+            delete object;
+            return nullptr;
+        }
 
-        FPType * maxs = tlsMaxs.local();
-        DAAL_CHECK_MALLOC(maxs);
+        bool isValid() const { return _buff.get(); }
 
-        FPType * kDistances = tlsKDistances.local();
-        DAAL_CHECK_MALLOC(kDistances);
+    private:
+        BruteForceTask(const size_t inBlockSize, const size_t outBlockSize, const size_t k)
+        {
+            size_t kDistancesSize = inBlockSize * k * sizeof(FPType);
+            size_t maxsSize       = inBlockSize * sizeof(FPType);
+            size_t kIndexesSize   = inBlockSize * k * sizeof(int);
 
-        int * kIndexes = tlsKIndexes.local();
-        DAAL_CHECK_MALLOC(kIndexes);
+            _buff.reset(kDistancesSize + maxsSize + kIndexesSize);
 
-        int * voting = tlsVoting.local();
-        DAAL_CHECK_MALLOC(voting);
+            kDistances = reinterpret_cast<FPType *>(&_buff[0]);
+            maxs       = reinterpret_cast<FPType *>(&_buff[kDistancesSize]);
+            kIndexes   = reinterpret_cast<int *>(&_buff[kDistancesSize + maxsSize]);
 
-        service_memset_seq<FPType, cpu>(maxs, MaxVal<FPType>::get(), trainBlockSize);
-        service_memset_seq<FPType, cpu>(kDistances, MaxVal<FPType>::get(), blockSize * k);
+            service_memset_seq<FPType, cpu>(maxs, MaxVal<FPType>::get(), outBlockSize);
+            service_memset_seq<FPType, cpu>(kDistances, MaxVal<FPType>::get(), inBlockSize * k);
+        }
+
+        TArrayScalable<byte, cpu> _buff;
+    };
+
+    services::Status computeKNearestBlock(daal::algorithms::internal::EuclideanDistances<FPType, cpu> * distancesInstance, const size_t blockSize,
```

Comment:
Fix const for parameters where it does not follow consistency with the rest of the lib